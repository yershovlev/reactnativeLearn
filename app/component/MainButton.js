import React from 'react';
import {StyleSheet, Text, TouchableOpacity } from 'react-native';

import colors from '../config/colors'

function MainButton({ title, color = "primary", onPress }) {
    return (
        <TouchableOpacity style={[styles.button, { backgroundColor : colors[color],}]}
                          onPress={onPress}>
            <Text style={styles.buttonLable}> {title} </Text>
        </TouchableOpacity>
    );
}

export default MainButton;

const styles = StyleSheet.create({
    button: {
        borderRadius : 25,
        alignItems: 'center',
        justifyContent: 'center',
        padding : 15,
        width : '100%',
        marginVertical: 10,
    },

    buttonLable: {
        color : colors.white,
        fontSize : 18,
        textTransform : 'uppercase',
        fontWeight: 'bold',
    }
})