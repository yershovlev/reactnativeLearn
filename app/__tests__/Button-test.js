import React from "react";
import renderer from "react-test-renderer";
import MainButton from "../component/MainButton";

test("renders correctly", () => {
  const tree = renderer.create(<MainButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
