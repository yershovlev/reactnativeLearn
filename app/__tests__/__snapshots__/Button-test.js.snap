// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "green",
      "borderRadius": 25,
      "justifyContent": "center",
      "marginVertical": 10,
      "opacity": 1,
      "padding": 15,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "bold",
        "textTransform": "uppercase",
      }
    }
  >
     
     
  </Text>
</View>
`;
